raft/src/main/java/org/apache/kafka/snapshot/RecordsSnapshotWriter.java

```diff
@@ -237,12 +238,12 @@ public <T> RecordsSnapshotWriter<T> build(RecordSerde<T> serde) {
                             .setLastContainedLogTimestamp(lastContainedLogTimestamp)
                     );
 
-                    if (kraftVersion > 0) {
+                    if (kraftVersion != KRaftVersion.KRAFT_VERSION_0) {
```

Comment:
How about moving this predicate to `KRaftVersion::isReconfigSupported`? I think we are going to use this predicate in a few places.